mediafile.py

```diff
@@ -462,17 +462,22 @@ def __init__(self, key, as_type=six.text_type, suffix=None,
         - `float_places`: When the value is a floating-point number and
           encoded as a string, the number of digits to store after the
           decimal point.
+        - `read_only`: When true, sets the store method to a callable
+          which does nothing and returns None.
         """
         self.key = key
         self.as_type = as_type
         self.suffix = suffix
         self.float_places = float_places
-
+        self.read_only = read_only
         # Convert suffix to correct string type.
         if self.suffix and self.as_type is six.text_type \
            and not isinstance(self.suffix, six.text_type):
             self.suffix = self.suffix.decode('utf-8')
 
+        if self.read_only:
+            self.store = lambda *args, **kwargs: None
```

Comment:
I understand the desire to centralize the logic, but overwriting the method is a little clever and may be non-obvious to maintain. It would be great if we could just have a `if not self.readonly:` check in the `store` or `set` method, but of course there are many different overriding versions of this method.

Maybe the clean thing would be to introduce a new method that calls `set` conditionally (and which is never overridden)?

Reply:
I rewrote it to use conditional statements in the __set__ methods of `MediaField` and its descendants. I also took out the tag as requested below and will put it in another pull request once this is completed, and rewrote the test so that it creates/adds a read only field dynamically.